pkg/controller/direct/dataform/repository_controller.go

```diff
@@ -191,16 +190,43 @@ func (a *Adapter) Update(ctx context.Context, u *unstructured.Unstructured) erro
 	updateMask := &fieldmaskpb.FieldMask{}
 
 	if a.desired.Spec.GitRemoteSettings != nil {
-		if !reflect.DeepEqual(a.desired.Spec.GitRemoteSettings, a.actual.GitRemoteSettings) {
+		mapCtx := &direct.MapContext{}
+		protoDesired := RepositoryGitRemoteSettings_ToProto(mapCtx, a.desired.Spec.GitRemoteSettings)
+		if mapCtx.Err() != nil {
+			return fmt.Errorf("converting GitRemoteSettings to api: %w", mapCtx.Err())
+		}
+
+		if !reflect.DeepEqual(protoDesired, a.actual.GitRemoteSettings) {
```

Comment:
This bit (the reflect.DeepEqual) does look good though.  We might also be able to use `proto.Equal`